src/Platform/PlatformWindow.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <string>
+#include <memory>
+
+#include "Utility/Geometry/Size.h"
+#include "Utility/Geometry/Point.h"
+
+#include "PlatformOpenGLOptions.h"
+
+class PlatformOpenGLContext;
+
+/**
+ * Abstraction for accessing platform-specific window API.
+ *
+ * This class is essentially an API handle and doesn't hold any state. Windows users can think about it as
+ * a thin wrapper atop of `HWND`, so a name like `PlatformWindowApiHandle` might've been more suitable here.
```

Comment:
there's nothing wrong with the longer names as they are more descriptive and allow you to spend less time understanding, basically reading the code as it was a book
Java's problem is that it forces you to use nice names and then create monstrosities like PlatformWindowApiHandle platformWindowApiHandle = new PlatformWindowApiHandle()

That's not nice, i agree
Fortunately in c++ it's a bit batter :  auto window = std::make_shared<PlatformWindowApiHandle()

Reply:
Ahahaa, I'm in the "almost always auto is almost always a bad idea" camp, so I spell out my types except for iterators, lambdas, and structured bindings.

I've worked on several codebases that followed Almost Always Auto, was a constant pain figuring out wtf was going on.